### PR TITLE
Reenable Travis builds for MacOS

### PR DESCRIPTION
**Pull request by jamescowens**
This makes adjustments to .travis.yml and depends packages to reenable continuous integration builds. Note that the MacOS build, like all the builds other than the main Linux x86_64 non-depends build, is done with Qt off. For the other builds, this is done because of the 50 min Travis timeout. For MacOS it is for that reason, and also because the SVG module in Qt doesn't compile with the 10.11 SDK currently.

This is better than nothing for now. We really need to move to a container based build approach similar to current Bitcoin.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ env:
     - CCACHE_TEMPDIR=/tmp/.ccache-temp
     - CCACHE_COMPRESS=1
     - BASE_OUTDIR=$TRAVIS_BUILD_DIR/out
-    - SDK_URL=https://bitcoincore.org/depends-sources/sdks
+    - SDK_URL=https://github.com/phracker/MacOSX-SDKs/releases/download/10.13
```
Comment (cyrossignol) [inline .travis.yml]: From what I read in issue # 13 in that repository, we cannot download and compile against these SDKs in the cross-compiling environment currently set up for CI:

> You are expressly prohibited from separately using the Apple SDKs or attempting to run any part of the Apple Software on non-Apple-branded hardware.

_[XCode SDK Agreement](https://www.apple.com/legal/sla/docs/xcode.pdf)_

We may need to run CI for macOS builds on [Travis' macOS builders](https://docs.travis-ci.com/user/reference/osx).

Comment (cyrossignol) [reply .travis.yml]: I'm not sure if this language only applies to running the final software produced by building against the SDKs, or if it prohibits compilation against the SDKs on non-Apple hardware. FWIW, Bitcoin provides version 10.11 here: https://bitcoincore.org/depends-sources/sdks/MacOSX10.11.sdk.tar.gz

Comment (jamescowens) [reply .travis.yml]: Let's switch to Bitcoin's image. I trust it more. I have looked at that agreement before. Paragraph 2.2A ii. appears to allow the use of MacOS SDK's to develop applications, as it is excluded from the more strict requirements around general Apple SDK's.

Comment (jamescowens) [reply .travis.yml]: image

Comment (jamescowens) [reply .travis.yml]: The language is somewhat confusing, but it clearly carves out the MacOS SDK from the Apple SDK in general and allows more permissive use of the MacOS SDK. Unless I am mistaken, that is the only thing we are using.

And... I regard Travis CI VM's that are expressely set up for testing purposes such as ours, as meeting paragraph A's initial conditions.

Comment (jamescowens) [reply .travis.yml]: And... you are right. We need to switch to the Travis MacOS VM's for Mac building...

Comment (jamescowens) [reply .travis.yml]: But that can come later.

Comment (cyrossignol) [reply .travis.yml]: I agree with your conclusions.

**Review by cyrossignol — COMMENTED** [review]
The x64_64-apple-darwin11 cross-compilation works, but could not get the native_cdrkit package to compile with the new bzip2 package in depends. I needed to install the distribution's library. Is this something we want to make work?

**Comment by jamescowens** [comment]
Yes, but we can push that off until later. We are not using Travis, or depends for that matter, to actually build a working release. Currently it is only for compile sanity checking. Given that I think what we have here is sufficient for now.

**Review by cyrossignol — APPROVED** [review]
Sounds good to me.

**Pull request merged**